File: apps/admin_mobile/lib/domain/entities/availability.dart
import 'package:equatable/equatable.dart';

class AvailabilitySlot extends Equatable {
  final int id;
  final int dayOfWeek;
  final String startTime;
  final String endTime;
  final int maxConcurrent;
  final bool isActive;

  const AvailabilitySlot({
    required this.id,
    required this.dayOfWeek,
    required this.startTime,
    required this.endTime,
    required this.maxConcurrent,
    required this.isActive,
  });

  @override
  List<Object?> get props => [id];
}

class AvailabilityBlock extends Equatable {
  final int id;
  final String date;
  final String? startTime;
  final String? endTime;
  final String? reason;
  final DateTime createdAt;

  const AvailabilityBlock({
    required this.id,
    required this.date,
    this.startTime,
    this.endTime,
    this.reason,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id];
}

File: apps/admin-v2/src/domain/entities/availability.ts
export interface AvailabilitySlot {
  id: string;
  dayOfWeek: number;
  startTime: string;
  endTime: string;
  maxConcurrent: number;
  isActive: boolean;
}

export interface AvailabilityBlock {
  id: string;
  date: string;
  startTime: string | null;
  endTime: string | null;
  reason: string | null;
  createdAt: Date;
}
